src/PyNvCodec/src/PyNvEncoder.cpp

```diff
@@ -352,6 +374,66 @@ bool PyNvEncoder::EncodeSurface(shared_ptr<Surface> rawSurface,
   return EncodeSingleSurface(ctx);
 }
 
+bool PyNvEncoder::EncodeFromNVCVImage(py::object& nvcvImage,
+                                py::array_t<uint8_t>& packet)
+{
+  struct NVCVImageMapper {
+    int nWidth[2];
+    int nHeight[2];
+    int nStride[2];
+    CUdeviceptr ptrToData[2];
+  };
+
+  NVCVImageMapper nv12Mapper;
+
+  memset(&nv12Mapper, 0, sizeof(NVCVImageMapper));
+  py::list list = nvcvImage.attr("cuda")();
+
+  for (size_t i = 0; i < list.size(); i++) {
+    py::object nvcvImageType = list[i];
+
+    if (py::hasattr(nvcvImageType, "__cuda_array_interface__")) {
+      py::dict dict =
+          nvcvImageType.attr("__cuda_array_interface__").cast<py::dict>();
+      if (!dict.contains("shape") || !dict.contains("typestr") ||
+          !dict.contains("data") || !dict.contains("version")) {
+        return false;
```

Comment:
Does CVCUDA make use of the `stream` attribute of `__cuda_array_interface__` v3?